jade/wsgi/routes/v1/judgements.py

```diff
@@ -2,49 +2,78 @@
 
 import flask
 
-from func_tools import wraps
+from ... import responses, util
+from .... import types
 
-from ... import responses
 
+def configure(bp, config, trusted_clients, centralauth, state, stream):
 
-def configure(bp, trusted_clients, source):
+    @bp.route("/v1/judgements/<string:context>", methods=["get"])
+    def get_context_judgements(context):
+        """Gets all judgements within a context"""
+        judgements_doc = state.judgements.get(context)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<string:context>/<string:type>", methods=["get"])
+    def get_entity_type_judgements(context, type_):
+        """Gets all judgements for a specific entity type"""
+        judgements_doc = state.judgements.get(context, type_)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<string:context>/<string:type>/<int:id>", methods=["get"])  # noqa
+    def get_entity_judgements(context, type_, id_):
+        """Gets all judgements for a specific entity"""
+        judgements_doc = state.judgements.get(
+            context, type_, id_)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<string:context>/<string:type>/<int:id>/<string:schema>", methods=["get"])  # noqa
+    def get_entity_schema_judgements(context, type_, id_, schema):
+        """Gets all judgements for a specific entity schema"""
+        judgements_doc = state.judgements.get(
+            context, type_, id_, schema)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<int:judgement_id>/preference", methods=["put"])
+    @util.authorized_user_action(
+        config, trusted_clients, centralauth, "set_judgement_preference")
+    def set_judgement_preference(judgement_id, gu_id, request_values):
+        """
+        Sets the preference bit for a specific judgement and returns the event
+        """
+        preference = json.loads(request_values['preference'])
```

Comment:
This could also fail to be well-formed json, what would happen in that case?  I think it throws a ValueError, which we should catch.

Reply:
Right, I don't do all the error handling when we're here.  I think we might want to consider adding a decorator that will handle many error types and return a specific type of structured response.  I haven't worked out all of the details for that yet.  I'm not sure it will make sense (DRY, KISS) until it is half-implemented.

Reply:
That sounds great.

Reply:
Updates made!  I put the logic for formatting an exc in the Exception classes themselves.  I'm not sure how I feel about that.  What do you think?

Reply:
It looks really nice to use once implemented!